### ISSUE TITLE
[APP] GMail - email search

### ISSUE THREAD

**Issue by ad-m** [PipedreamHQ/pipedream#1910]
**Name of app / service**

GMail

**Link to developer documentation**

https://developers.google.com/gmail/api
**Is lack of support preventing you from building workflows, or do you have a workaround?**

It prevents me from building workflow.

**Are there specific actions, or triggers, you'd like to see for this app? Please let us know here or use the Action and Trigger issue templates to open requests for each!**

I would like periodically delete all emails with the label "CI" from my mailbox. I would like to periodically delete all e-mail messages that are older than x days and labeled "continous-integration". I can use Google AppScripts for this, but whoever uses it knows how unpleasant it is to use long-term (UI and UX horrible). GMail filters do not allow operations over time as evaluated only on ingress.

**Comment by dylburger** [comment]
@ad-m Thank you very much for the request. I have used AppsScript and @dannyroosevelt and I built a startup on email productivity before Pipedream, so the email deletion use case completely resonates.

We are tracking at #182 if you'd like to follow there. This issue will automatically be linked to that one, so when we build the integration, we'll note the deletion use case (which requires a special scope from Google).
